projects/slap_manipulation/src/slap_manipulation/dataloaders/peract_loader.py

```diff
@@ -0,0 +1,582 @@
+            new_data["proprio"] = data["all_proprio"]
+
+            # add language signals
+            new_data["cmd"] = data["cmd"]
+            description = data["cmd"]
+            new_data["gripper_width_array"] = data["gripper_width_array"]
+            new_data["all_proprio"] = data["all_proprio"]
+            # tokens = clip.tokenize([description]).numpy()
+            # token_tensor = torch.from_numpy(tokens).to(self.device)
+            # lang, lang_emb = self._clip_encode_text(token_tensor)
+            # lang = self.clip_encode_text(description)
+            # new_data["cmd_embs"] = lang
+
+            # discretize supervision signals
+            (
+                trans_action_indices,
+                grip_rot_action_indices,
+                ignore_colls,
+                gripper_states,
+                attention_coords,
+            ) = ([], [], [], [], [])
+            for idx in range(len(data["global_proprio"])):
+                (
+                    trans_action_index,
+                    grip_rot_action_index,
+                    ignore_coll,
+                    gripper_state,
+                    attention_coord,
+                ) = self._get_action(new_data, 5, idx, len(data["global_proprio"]))
+                trans_action_indices.append(trans_action_index)
+                grip_rot_action_indices.append(grip_rot_action_index)
+                ignore_colls.append(ignore_coll)
+                gripper_states.append(gripper_state.unsqueeze(0))
+                attention_coords.append(attention_coord)
+            trans_action_indices = torch.cat(trans_action_indices)
+            grip_rot_action_indices = torch.cat(grip_rot_action_indices)
+            ignore_colls = torch.cat(ignore_colls)
+            gripper_states = torch.cat(gripper_states, dim=0)
+            attention_coords = torch.cat(attention_coords)
+
+            # # add discretized signal to dictionary
+            # # concatenate gripper ori and pos
+            gripper_pose = torch.FloatTensor(data["global_proprio"][:, :-1])
+            new_data.update(
+                {
+                    "trans_action_indices": trans_action_indices,
+                    "rot_grip_action_indices": grip_rot_action_indices,
+                    "ignore_collisions": ignore_colls,
+                    "gripper_pose": gripper_pose,
+                    "gripper_states": gripper_states,
+                    "attention_coords": attention_coords,
+                }
+            )
+
+            # TODO make sure you update the keys used in peract_model.update() method
+            # preprocess and normalize obs
+            rgb, pcd = self._preprocess_inputs(data)
+            new_data["rgb"] = rgb
+            new_data["xyz"] = pcd
+            new_data["action"] = gripper_states
+
+            if self.is_action_valid(new_data):
+                return new_data
+            else:
+                tries += 1
+
+        return new_data
+
+    def _norm_rgb(self, x):
```

Comment:
normalize RGB - RGB is expected to be between 0 and 1?? add a sanity check here?